randovania/patching/prime/asset_conversion.py

```diff
@@ -127,52 +150,130 @@ def id_generator(asset_type):
         next_id = result + 1
         return result
 
-    try:
-        asset_provider = prime_asset_provider()
-    except RuntimeError:
-        return
-
     updaters = status_update_lib.split_progress_update(status_update, 3)
+    cache_path = Options.with_default_data_dir().internal_copies_path.joinpath("prime2", "prime1_models")
+
+    if get_asset_cache_version(RandovaniaGame.METROID_PRIME_ECHOES, RandovaniaGame.METROID_PRIME) >= ECHOES_MODEL_VERSION:
+        print("Reading assets from cache")
+        converted_assets = {}
+        # Read assets from cache if available
+        for asset_path in cache_path.glob("*"):
+            if asset_path.suffix[1:].upper() not in conversions.ALL_FORMATS:
+                continue
+            type = asset_path.suffix[1:]
+            asset_id = int(asset_path.stem)
+
+            construct_class = format_for(type)
+
+            raw = asset_path.read_bytes()
+            decoded_from_raw = construct_class.parse(raw, target_game=Game.ECHOES)
+            converted_asset = ConvertedAsset(
+                id=asset_id,
+                type=type,
+                resource=decoded_from_raw,
+            )
+            converted_assets[asset_id] = converted_asset
+
+        with open(cache_path.joinpath("meta.json"), "r")as data_additions_file:
+            randomizer_data_additions = json.load(data_additions_file)["data"]
+
+
+    else:
+        print("Creating assets....")
+
+        try:
+            asset_provider = prime_asset_provider()
+        except RuntimeError:
```

Comment:
Now that we can properly prompt the user for the ISO, this should be a fatal error